fem/fespace.cpp

```diff
@@ -728,19 +866,100 @@ void FiniteElementSpace::BuildConformingInterpolation() const
             list.OrientedPointMatrix(slave, T.GetPointMat());
-            fe->GetLocalInterpolation(T, I);
+            slave_fe->GetTransferMatrix(*master_fe, T, I);
+
+            // variable order spaces: face edges need to be handled separately
+            int skipfirst = 0;
+            if (IsVariableOrder() && entity == 2 && slave.index >= 0)
+            {
+               skipfirst = GetNumBorderDofs(master_geom, q);
+            }
 
             // make each slave DOF dependent on all master DOFs
-            AddDependencies(deps, master_dofs, slave_dofs, I);
+            AddDependencies(deps, master_dofs, slave_dofs, I, skipfirst);
+
+            if (skipfirst)
+            {
+               // constrain internal edge DOFs if they were skipped
+               const auto *pm = list.point_matrices[master_geom][slave.matrix];
+               AddEdgeFaceDependencies(deps, master_dofs, master_fe,
+                                       slave_dofs, slave.index, pm);
+            }
+         }
+
+         // Add inverse dependencies for the cR_hp matrix; if a master has
+         // more DOF sets, the lowest order set interpolates the highest one.
+         if (IsVariableOrder())
+         {
+            int nvar = GetNVariants(entity, master.index);
+            if (nvar > 1)
+            {
+               int q = GetEntityDofs(entity, master.index, highest_dofs,
+                                     master_geom, nvar-1);
+               const auto *highest_fe = fec->GetFE(master_geom, q);
+
+               T.SetIdentityTransformation(master_geom);
+               master_fe->GetTransferMatrix(*highest_fe, T, I);
+
+               // add dependencies only for the inner dofs
+               int skip = GetNumBorderDofs(master_geom, p);
+               AddDependencies(inv_deps, highest_dofs, master_dofs, I, skip);
+            }
+         }
+      }
+   }
+
+   // variable order spaces: enforce minimum rule on conforming edges/faces
+   if (IsVariableOrder())
+   {
+      for (int entity = 1; entity < mesh->Dimension(); entity++)
+      {
+         const Table &ent_dofs = (entity == 1) ? var_edge_dofs : var_face_dofs;
+         int num_ent = (entity == 1) ? mesh->GetNEdges() : mesh->GetNFaces();
+         MFEM_ASSERT(ent_dofs.Size() == num_ent+1, "");
+
+         // add constraints within edges/faces holding multiple DOF sets
+         Geometry::Type last_geom = Geometry::INVALID;
+         for (int i = 0; i < num_ent; i++)
+         {
+            if (ent_dofs.RowSize(i) <= 1) { continue; }
+
+            Geometry::Type geom =
+               (entity == 1) ? Geometry::SEGMENT : mesh->GetFaceGeometry(i);
+
+            if (geom != last_geom)
+            {
+               T.SetIdentityTransformation(geom);
+               last_geom = geom;
+            }
+
+            // get lowest order variant DOFs and FE
+            int p = GetEntityDofs(entity, i, master_dofs, geom, 0);
```

Comment:
Why `GetEntityDofs` is returning the order `p`?

Reply:
It's the actual order of the variant. It's needed for `GetFE` in subsequent code. This is very internal, why does it bother you?

Reply:
I'm just trying to understand the code...